src/init.cpp

```diff
@@ -70,9 +71,20 @@ struct Init<Backend::MC> {
         cfg.umpire_host_memory_pool_initial_block_bytes, cfg.umpire_host_memory_pool_next_block_bytes,
         cfg.umpire_host_memory_pool_alignment_bytes, cfg.umpire_host_memory_pool_coalescing_free_ratio,
         cfg.umpire_host_memory_pool_coalescing_reallocation_ratio);
+    // install mpi polling loop
+    int mpi_initialized;
+    DLAF_MPI_CHECK_ERROR(MPI_Initialized(&mpi_initialized));
+    if (mpi_initialized) {
+      pika::mpi::experimental::start_polling(pika::mpi::experimental::exception_mode::no_handler);
+    }
```

Comment:
Sorry for not noticing this before. Perhaps this could be moved into the if in https://github.com/eth-cscs/DLA-Future/blob/e9a661d7190c2c27841d341ac5b8bae38b60abfb/src/init.cpp#L374-L381? Since the MPI init isn't specific to neither `Backend::MC` nor `Backend::GPU`.

Same for stopping the polling?